### PR TITLE
[RLlib] Initial code/comment cleanups in preparation for decentralized multi-agent learner.

### PR DESCRIPTION
**Pull request by sven1977**
Initial code/comment cleanups in preparation for decentralized multi-agent learner.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/agents/trainer.py [hunk 1]
```diff
@@ -1295,6 +1301,12 @@ def training_iteration(self) -> ResultDict:
         else:
             train_results = multi_gpu_train_one_step(self, train_batch)
 
+        # Update weights - after learning on the local worker - on all remote
+        # workers.
+        if self.workers.remote_workers():
```
Comment (gjoliver) [inline rllib/agents/trainer.py:1306]: do we always sync weights after each iteration?
for off policy methods, how often weights get synced feels like a hyper-param?

File: rllib/execution/train_ops.py [hunk 2]
```diff
@@ -55,13 +55,6 @@ def train_one_step(trainer, train_batch) -> Dict:
     trainer._counters[NUM_ENV_STEPS_TRAINED] += train_batch.count
     trainer._counters[NUM_AGENT_STEPS_TRAINED] += train_batch.agent_steps()
 
-    # Update weights - after learning on the local worker - on all remote
```
Comment (gjoliver) [inline rllib/execution/train_ops.py:58]: actually why do we move this into Trainer?

**Review by gjoliver — COMMENTED** [review]
this is super nice clean up change!!
I have a couple of really simple questions. thanks.

File: rllib/agents/trainer.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/trainer.py:1306]: Remember, this is only the basic, default iteration function. Off-policy algos will define a different one.

Comment (sven1977) [reply rllib/agents/trainer.py:1306]: Just wanted to move this out of the `train_one_step` function, as I felt weight synching is kind of it's own logical unit within an iteration.

File: rllib/execution/train_ops.py (hunk 2 above)
Comment (sven1977) [reply rllib/execution/train_ops.py:58]: More logical to separate these two concepts:
- train one step
- update weights

**Review by gjoliver — APPROVED** [review]
ok, get it.

**Comment by gjoliver** [comment]
you probably want to merge master to fix the lint issue.
thanks for the clean up.

**Pull request merged**